src/Expander.hs

```diff
@@ -492,10 +494,10 @@ initializeKernel = do
             modifyState $ over (expanderDefTypes . at ph . non Env.empty) $
               Env.insert var x schPtr
             forkGeneralizeType exprDest t schPtr
-            nowValidAt pdest (SpecificPhase p)
+            linkDeclValidity vp (ScopeSet.singleScopeAtPhase sc p)
```

Comment:
Instead of filling the `DeclValididtyPtr` with a `PhaseSpec` (which no longer exists), we now fill it with an equivalent `ScopeSet`. For `SpecificPhase p`, the equivalent `ScopeSet` is one which assigns the scope to phase `p`.

Reply:
I had switched from `SpecificPhase p` to `ScopeSet.singleScopeAtPhase sc p` and then to `addPhase p scs sc` (or equivalently `scs <> ScopeSet.singleScopeAtPhase sc p`), and now I'm back to `ScopeSet.singleScopeAtPhase sc p` because I found a way to accumulate multiple scopes without forcing each declaration form to thread through a `ScopeSet`. Now each declaration form only needs to fill their `DeclValidityPtr` with the scopes they introduce.